### DISCUSSION TITLE
HocusPocus Provider onClose stating there's no token

### DISCUSSION THREAD

**Discussion by ricotrevisan** [ueberdosis/hocuspocus#809] [Q&A]
When a client idles for a minute or so, this message pops up and disconnects the client.

```js
HocuspocusProviderWebsocket.ts:495 [HocuspocusProvider] An authentication token is required, but you didn’t send one. Try adding a `token` to your HocuspocusProvider configuration. Won’t try again.
onClose @ HocuspocusProviderWebsocket.ts:495
(anonymous) @ EventEmitter.ts:19
emit @ EventEmitter.ts:19
onCloseHandler @ HocuspocusProviderWebsocket.ts:307
```

The client is sending a token, the server is authenticating it, so I'm not sure where this issue is coming from. Any ideas what could be happening? Do I have to set something to the payload on the onAuthenticate hook?

**Comment by adschwartz** [comment]
I'm having this problem too. I'm also providing the token and don't see why I would get this message.
